### ISSUE TITLE
Check / Change Units in p2_FORESCE_LC_per_catchment_reclass_cat$total_PRMS_area

### ISSUE THREAD

**Issue by msleckman** [USGS-R/drb-inland-salinity-ml#108]
Change to SQKM as NLCD is in that area at the moment  

drb-network-prep PR#2 [comment](https://github.com/USGS-R/drb-network-prep/pull/2#discussion_r821211020)

**Comment by jds485** [comment]
@msleckman : did you address this issue in #97? Can it be closed?

**Comment by msleckman** [comment]
Not addressed in #97. Should be done when combining NLCD and FORESCE.

**Comment by msleckman** [comment]
PRMS area from edited shapefile are different from the PRMS area given in the NLCD tables. This might be due to rounding process that occurs when aggregating from comid to PRMS in the original NLCD proportions datasets

```
p1_catchments_edited_sf_areakm2 <- p1_catchments_edited_sf %>%
  sf::st_drop_geometry() %>% group_by(PRMS_segid) %>%
  summarise(PRMS_area_km2 = sum(hru_area_m2)/10^6)

summary(p1_catchments_edited_sf_areakm2$PRMS_area_km2)
    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
  0.0222  20.5840  55.5049  78.3554 112.0621 541.9462 
```

vs: 

```
> summary(p2_PRMS_NLCD_lc_proportions_reclass_cat[[1]]$AREASQKM_PRMS)
    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
  0.0009  13.0510  44.2323  70.9794 104.5773 538.9892
```

- [ ] Check proportion_lc_by_prms_tot$AREASQKM_PRMS. Think it is the area of catchment only, not Total upstream. 
Need to change `proportion_lc_by_prms.R` and add a parameter that accepts the `TOTDASQKM` colname as well as `AREASQKM` 

This blocks #102 because the PRMS areas in final FORESCE (`p2_FORESCE_lc_proportions_reclass_cat`) != NLCD areas (`p2_PRMS_NLCD_lc_proportions_reclass_cat`)

**Comment by msleckman** [comment]
Produced [gist](https://gist.github.com/msleckman/e10719e4396bd7dc3b8a89ec8657994c) to review and plot the area difference between area in FORESCE dataset (derived from edited catchments shapefile) and area given in processed NLCD data tables and scaled from comid/nhd to PRMS. Review and comparison done on the back of similar issue #45.


![image](https://user-images.githubusercontent.com/36547359/160631956-f3c89f0d-1110-4b57-a7d1-137941a870d4.png)

![image](https://user-images.githubusercontent.com/36547359/160632075-e6805687-fc96-4ccb-8c88-1c3d518ef8c8.png)

We find about ~28 PRMS segments have catchment polygon area in NLCD != catchment polygon area in FORESCE by over 50 km2. About 17 PRMS segments have a difference over 10 km2. 

sample: 
<img width="713" alt="image" src="https://user-images.githubusercontent.com/36547359/160633131-bd2f49b5-d297-4742-87c3-bbc0617ab5c7.png">

<img width="713" alt="image" src="https://user-images.githubusercontent.com/36547359/160633244-95ffd056-92cf-4bc5-93d6-7a10baad2a0b.png">

TBC:
- [ ] complete unchecked bullet above. 
- [ ] Examine source of difference

**Comment by msleckman** [comment]
Branch 108-... addressing this in my forked repo. 

Added function [area_diff_fix.R](https://github.com/msleckman/drb-inland-salinity-ml/blob/108-check-change-units-in-p_foresce_lc_per_catchment_reclass_cat%24total_prms_area/2_process/src/area_diff_fix.R) with work in progress on extracting the PRMS_segid with area discrepancy and fixing by pulling it's attributed nhd catchments.

**Comment by msleckman** [comment]
> I addressed Change FORE-SCE to SQKM as NLCD is in that area at the moment in https://github.com/USGS-R/drb-inland-salinity-ml/issues/126 by adding hru_area_km2 to p1_catchments_edited_sf. It sounds like you're also using this issue to make the FORE-SCE and NLCD catchment areas match, so maybe https://github.com/USGS-R/drb-inland-salinity-ml/issues/126 shouldn't close this issue

Ah, did not see this. Hopefully this doesnt complicate the pipeline too much (it shouldnt) as I also addressed this in #134 PR that I just submit. I think this PR makes the task transfer smoother.
